caigo.go

```diff
@@ -90,12 +90,14 @@ func (sc StarkCurve) Sign(msgHash, privKey *big.Int, seed ...*big.Int) (x, y *bi
 	}
 
 	invalidK := true
+	inSeed := big.NewInt(0)
+
 	for invalidK {
-		inSeed := big.NewInt(0)
 		if len(seed) == 1 {
 			inSeed = seed[0]
 		}
 		k := sc.GenerateSecret(new(big.Int).Set(msgHash), new(big.Int).Set(privKey), inSeed)
+		inSeed = inSeed.Add(inSeed, big.NewInt(1))
```

Comment:
Let's add a quick in-line comment about why it's important to increment the seed here

Reply:
Done